18_links_lab/linkslab.txt

```diff
@@ -1,20 +1,59 @@
 Preparation: Perform the following  (no need to write the commands you used):
 
-  - create a new directory named "links"
+ 
+ - create a new directory named "links"
   - cd to it
-  - inside the directory create 2 text files named "file1" and "file2". Write the names of some things you like in file "file1", and some things you don't like in file "file2"
+  
+
+mkdir links
+ln -s links  roi
+
+- inside the directory create 2 text files named "file1" and "file2". Write the names of some things you like in file "file1", and some things you don't like in file "file2"
+
 
+
+cd links
+pico file1
+pico file2
+
 1. Perform the following and write the commands you used, from within links directory:
 
-  - Create a symbolic link named file3 that points to "file1"
-  Answer: 
+  
+
+- Create a symbolic link named file3 that points to "file1"
+ 
+Answer: 
 
-  - Create a hard link named file4 that points to "file2"
-  Answer: 
+ ln -s file 1 file 3
+
+ - Create a hard link named file4 that points to "file2"
+
+  Answer: ln "file2" "file4"
+
+ 
 
   - Create a new directory named "tmp" in your home folder:
-  Answer:
+ 
+ Answer:
 
-2. Move "file3" to the new "~/tmp" directory you just created. Can you still use it to see the original "file1" ? Why not? How would you create a symbolic link that would survive the transfer to ~/tmp ?
+ 
+
+
+2. Move "file3" to the new "~/tmp" directory you just created. Can you still use it to see the original "file1" ? Why not? How would you create a symbolic link that would survive the transfer to ~/tmp ?
+
+
+mv links/file3 ~/tmp
+cd ~/tmp
+ls -l file3
+file3 ---> file1
+ls ../file3
+
+no such file or directory. 
+
+ln-s /home/rpi/tmp/file3  link-to-main
+ls -l link-to-main
+link-to-main ---> /home/roi/tmp/file3
+
+
 
 3. How can you tell that "file2" and "file4" refer to the same file?
```

Comment:
The command `ls -li` prints a number called the inode of a file. Two files are hard links to one another if they have the same inode number